cron/monthly/user-report.js

```diff
@@ -198,25 +196,28 @@ const processBacker = async FromCollectiveId => {
         createdAt: { [Op.gte]: startDate, [Op.lt]: endDate },
         SubscriptionId: { [Op.ne]: null },
       },
+      status: {
+        [Op.and]: {
+          [Op.ne]: ORDER_STATUS.ERROR,
+          [Op.ne]: ORDER_STATUS.CANCELLED,
+        },
+      },
+      deletedAt: null,
     },
     include: [{ model: models.Subscription }],
   });
-  const ordersByCollectiveId = {};
-  orders.map(o => {
-    ordersByCollectiveId[o.CollectiveId] = o;
-  });
+  // group orders(by collective) that either don't have subscription or have active subscription
+  const ordersByCollectiveId = groupBy(orders.filter(o => !o.Subscription || o.Subscription.isActive), 'CollectiveId');
   const collectivesWithOrders = [];
   collectives.map(collective => {
-    collectivesWithOrders.push({
-      ...collective,
-      order: ordersByCollectiveId[collective.id],
-    });
-  });
-  collectivesWithOrders.sort((a, b) => {
-    if (get(a, 'order.totalAmount') > get(b, 'order.totalAmount')) return -1;
-    else return 1;
+    // It's only possible to have one order with subscription active Per Collective
+    const collectiveOrders = ordersByCollectiveId[collective.id] && ordersByCollectiveId[collective.id];
+    collectiveOrders &&
```

Comment:
@znarf that commented you've just marked me is outdated. I'm just pushing to `collectivesWithOrders` after validating`collectiveOrders`.